xrdp/xrdp_tconfig.c

```diff
@@ -199,6 +199,70 @@ tconfig_load_gfx_x264_ct(toml_table_t *tfile, const int connection_type,
     return 0;
 }
 
+static int tconfig_load_gfx_order(toml_table_t *tfile, struct xrdp_tconfig_gfx *config)
+{
+    /*
+     * This config loader is not responsible to check if xrdp is built with
+     * H264/RFX support. Just loads configurations as-is.
+     */
+
+    TCLOG(LOG_LEVEL_TRACE, "[codec]");
+
+    int h264_found = 0;
+    int rfx_found = 0;
+
+    config->codec.h264_idx = -1;
+    config->codec.rfx_idx = -1;
+
+    toml_table_t *codec = toml_table_in(tfile, "codec");
+    toml_array_t *order = toml_array_in(codec, "order");
```

Comment:
Need to check codec against NULL.

Reply:
If either is NULL, I want to return default priority (jump to line 248) without using goto statement. My first code used goto but I've rewritten without goto.

I think it can be written like this. What do you think?

```C
toml_table_t *codec;
toml_array_t *order;

 if ((codec = toml_table_in(tfile, "codec")) &&
         (order = toml_array_in(codec, "order")))
```

Reply:
That's fine.

You can make the test explicit with :-

```c
 if ((codec = toml_table_in(tfile, "codec") != NULL) &&
         (order = toml_array_in(codec, "order") != NULL))
```